src/python/WMCore/WebTools/Root.py

```diff
@@ -93,16 +93,17 @@ def access(self):
         if not rbytes:
             try:
                 # request.rfile.rfile.bytes_read is a custom CMS web
-                #  cherrypy patch not always available, hence the test
-                rbytes = (getattr(request.rfile, 'rfile', None)
-                        and getattr(request.rfile.rfile, "bytes_read", None)
-                        and request.rfile.rfile.bytes_read) or "-"
+                # cherrypy patch not always available, hence the test
+                if request.rfile.rfile.bytes_read:
+                    rbytes = request.rfile.rfile.bytes_read
```

Comment:
Alan, I don't think it is safe to assume that request may have rfile, and rfile may have bytes_read data. I think getattr is better approach. But what you probably need is the following:
```
rfile = None
if hasattr(request, "rfile"):
   rfile = request.rfile
  if hasattr(rfile, "rfile"):
     rfile = rfile.rfile
bytes_read = None
if rfile and hasattr(rfile, "bytes_read"):
   bytes_read = getattr(rfile, "bytes_read")
if bytes_read:
   rbytes = bytes_read
else:
   rbytes = "-"
```

I suggest to provide proper unit test to pass different structure of request, e.g. if it is real cherrypy request, if it is None, etc.

Reply:
That nested getattr only make things unclear. Given that we are inside a try/except block, we can just try to access that attribute directly, which will fall into the except block if it does not exist.